packages/idyll-document/src/utils/index.js

```diff
@@ -61,30 +61,13 @@ const evalExpression = (acc, expr) => {
           })
           .join('\n')
       }
-      return ${
-        // IIFE since only expressions are allowed in template strings
-        (() => {
-          try {
-            if (typeof eval(expr) !== 'function') return expr;
-
-            // if the source expression is a function
-            // it needs to be run inside a try...catch to avoid RTEs
-            return (...rest) => {
-              try {
-                return eval(expr)(...rest);
-              } catch (e) {
-                console.warn('ERROR evaluating:', expr);
-                console.warn(e);
-                return null;
-              }
-            }
-          } catch (e) {
-            return expr;
-          }
-        })()
-      };
+      return ${expr};
     })()
-  `);
+  `;
+
+  try {
+    return eval(e);
+  } catch (err) {}
```

Comment:
I split out the construction from the eval call for easier debugging. There have been various times when I needed to check what was actually being evaluated and this format makes that easier.

More importantly, removing the interior try catch means errors still make it to the console. I think our error bound will keep things from getting into a bad state but do we care about the console showing errors? Probably not but I wanted to be sure.

Reply:
I think thats okay - it is frustrating if a program is broken and it doesn't show any error messages

Reply:
👍